src/foam/util/Emails/EmailsUtility.java

```diff
@@ -47,6 +47,7 @@ public static void sendEmailFromTemplate(X x, User user, EmailMessage emailMessa
     // Add template name to templateArgs, to avoid extra parameter passing
     if ( ! SafetyUtil.isEmpty(templateName) ) {
       if ( templateArgs != null ) {
+        emailMessage.setTemplateName(templateName);
```

Comment:
Can you explain more how this is helpful?  Is it that the templateArgs are not visible anywhere? 
If the templateArgs were added to the email message then it would take care of itself.   The emailMessage could format itself, and also when viewing the emailMessage as a model you could see the templateArgs (key/values pairs).

Reply:
The message says it is to help with debugging.
If we stored the template name and args instead of the body, then the email messages could be much smaller.

Reply:
The basic idea here is that we have a recurring error message in the logs that says we are trying to send an email message that has no content, but we aren't able to tell which email is actually failing because we can't see the code path that it's coming from .. the easiest way to discern the code path would be to see the email template that is being used. In order to figure out the email template, we can either just log it .. which we could do here .. or we could add it to the email message model, so that emails that are stuck in the 'UNSENT' state will have more information on where they came from .. because right now we aren't seeing anything in production for the unsent messages (i.e. the message content + subject are completely empty). So which do you prefer?

Reply:
Looks like the template args has the template name in it .. so that's the easiest way plus we get more information about what we are trying to populate .. better solution all around.